h/services/bulk_annotation.py

```diff
@@ -0,0 +1,190 @@
+from typing import List, Union
+
+import sqlalchemy as sa
+from sqlalchemy.engine import Row
+from sqlalchemy.orm import Session
+from sqlalchemy.sql import Select
+
+from h.models import Annotation, AnnotationModeration, Group, User
+
+
+class BadDateFilter(ValueError):
+    """There is something wrong with the date filter provided."""
+
+
+class BadFieldSpec(ValueError):
+    """There is something wrong with the field you have specified."""
+
+
+def date_match(column, spec: dict):
```

Comment:
This was previously in a file in `db.py`. This mirrors a pattern we are starting to see in LMS where multiple services  need to build similar clauses in their search functions.

However there is only one consumer right now, so it can sit here.